src/ucs/sys/sock.c

```diff
@@ -887,19 +888,28 @@ ucs_status_t ucs_sock_port_from_string(const char *port_str, uint16_t *port)
     return UCS_OK;
 }
 
-int ucs_sockaddr_cmp(const struct sockaddr *sa1,
-                     const struct sockaddr *sa2,
-                     ucs_status_t *status_p)
+static int ucs_sockaddr_are_both_known_af(const struct sockaddr *sa1,
```

Comment:
IMO it's not practical (nor logical) to have this function checking 2 addresses. Finally it's more verbose and cumbersome, I would propose to have just a function checking single address
```
static int ucs_sockaddr_check_af(const struct sockaddr *sa)
{
    if (!ucs_sockaddr_is_known_af(sa)
        error_message
        return 0
    return 1
}

Reply:
IMO it's ok to have a function just for error message, but for a single argument
With 2 arguments it's not clear which one is failing (or both?), and you have to check again

Reply:
how about moving the error message to `ucs_sockaddr_is_known_af`?

Reply:
IMO it's not desirable, because this function is called from `uct_rdmacm_cm_ep_str`, and it basically checks whether IP address is initialized or not:
```
    if ((local_addr != NULL) && ucs_sockaddr_is_known_af(local_addr)) {
        ucs_sockaddr_str(local_addr, local_ip_port_str, UCS_SOCKADDR_STRING_LEN);
    } else {
        ucs_strncpy_safe(local_ip_port_str, "<invalid>", UCS_SOCKADDR_STRING_LEN);
    }
```
So personally I would propose to add just a simple static function with error message in sock.c, and replace all the invocations of `ucs_sockaddr_is_known_af` inside sock.c with this new function.
```
int ucs_sockaddr_check_known_af(const struct sockaddr *sa)
{
    int res = ucs_sockaddr_is_known_af(sa);
    if (!res) {
        ucs_error("<invalid address family>", ...)
    }
    return res;
}
And then replace all 7 places where ucs_sockaddr_is_known_af is used inside sock.c
```